From aa16cd82538befcb1230239728a8dc1ce9593c93 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 29 Jul 2026 22:52:56 +0100
Subject: [PATCH] overlays: Don't configure PIO pins as such

Although it seems sensible to select the PIO function for pins to be
used for PIO, the pio-rp1 driver takes care of that. With strict pinmux
mode enabled, allowing pinctrl to claim it for PIO on behalf of the
driver will interfere with the client drivers, which use gpiod to
claim the pins. By not setting the pin function, or by setting it to
"gpio" (*), there is no contention.

(*) In fact, unless some other pinconf change is required, such as
setting or disabling a pull, no pinctrl declaration is required.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/pwm-pio-overlay.dts    | 1 -
 arch/arm/boot/dts/overlays/ws2812-pio-overlay.dts | 1 -
 2 files changed, 2 deletions(-)

--- a/arch/arm/boot/dts/overlays/pwm-pio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/pwm-pio-overlay.dts
@@ -11,7 +11,6 @@
 		__overlay__ {
 			pwm_pio_pins: pwm_pio_pins@4 {
 				brcm,pins = <4>; /* gpio 4 */
-				function = "pio";
 				bias-disable;
 			};
 		};
--- a/arch/arm/boot/dts/overlays/ws2812-pio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ws2812-pio-overlay.dts
@@ -11,7 +11,6 @@
 		__overlay__ {
 			ws2812_pio_pins: ws2812_pio_pins@4 {
 				brcm,pins = <4>; /* gpio 4 */
-				function = "pio";
 				bias-disable;
 			};
 		};
